### PR TITLE
[Packaging] Update setup.py for Beta release

### PR DESCRIPTION
**Pull request by langong347**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #334
* #326
* #322
* __->__ #323
* #333

Summary:

- Avoid intermediary variable assignments that are optional
- Make `get_version` public
- Add missing `python 3.7` as a supported version
- Add group email distribution list
- Add development status as `Beta` according to [`Pythons Trove Classifiers`](https://martin-thoma.com/software-development-stages/)
- Bump `version` from alpha to beta

Test Plan:

- Install torchmultimodal in editable mode and test import
- Unit test (see CI)

```
$ conda create -n test python=3.7
$ conda activate test
$ cd <project_root>
$ pip install -e .
$ cd ../
$ python -c "import torchmultimodal;print(torchmultimodal)"
<module 'torchmultimodal' from '/data/home/langong/repos/ci/torchmultimodal/__init__.py'>

 /fsx/users/langong/conda/envs/test_binary_py38/lib/python3.8/site-packages/__editable__.torchmultimodal-0.1.0b0.pth
    /fsx/users/langong/conda/envs/test_binary_py38/lib/python3.8/site-packages/__editable___torchmultimodal_0_1_0b0_finder.py
    /fsx/users/langong/conda/envs/test_binary_py38/lib/python3.8/site-packages/torchmultimodal-0.1.0b0.dist-info/*

```

Differential Revision: [D39904706](https://our.internmc.facebook.com/intern/diff/D39904706)

### CODE REVIEW

**Review by pikapecan — COMMENTED** [review]
Curious why do we prefer free format sting instead of keep constants?

File: setup.py [hunk 1]
```diff
@@ -41,36 +41,32 @@ def read_requirements(file):
...
-
 if __name__ == "__main__":
 
     setup(
-        name=DISTNAME,
+        name="torchmultimodal",
         include_package_data=True,
-        packages=find_packages(exclude=EXCLUDES),
+        packages=find_packages(
+            exclude=("examples*", "test*")
```
Comment (ankitade) [inline setup.py:50]: should this become "tests" now?

File: setup.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def clean_html(raw_html):
     return cleantext
 
 
-def _get_version():
+def get_version():
     # get version string from version.py
     version_file = os.path.join(os.path.dirname(__file__), "version.py")
```
Comment (ankitade) [inline setup.py:22]: do we leave the version same as old one?

Comment (langong347) [reply setup.py:22]: You mean the version number in [`version.py`](https://github.com/facebookresearch/multimodal/blob/main/version.py#L17)? We can change it to beta release but until we decide to do a stable release the version number won't show up in PyPI.

File: setup.py (hunk 1 above)
Comment (langong347) [reply setup.py:50]: Yeah good catch, although `test*` can still capture `tests*` but let me change it anways.

**Comment by langong347** [comment]
@pikapecan >Curious why do we prefer free format string instead of keep constants?

Mainly to shorten the script but neither do I see an obvious reason defining those constants.

**Review by pikapecan — APPROVED** [review]

**Review by ankitade — APPROVED** [review]

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904706).

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904706).

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39904706).

**Pull request closed without merging**